tutorials/tour/_posts/2017-02-13-singleton-objects.md

```diff
@@ -10,65 +10,111 @@ num: 13
+A singleton object with the same name as a class is called a _companion object_. Conversely, the class is the object's companion class. The companion class and object can access each other's private members. Use a companion object for methods and values which are not specific to instances of the companion class.
+```
+import scala.math._
 
-A class and its companion object, if any, must be defined in the same source file. Like this:
+class Circle(val radius: Double) {
+  def area: Double = Circle.calculateArea(radius)
+}
 
-```tut
-class IntPair(val x: Int, val y: Int)
+object Circle {
+  def calculateArea(radius: Double): Double = Pi * pow(radius, 2.0)
+}
 
-object IntPair {
-  import math.Ordering
+val circle1 = new Circle(5.0)
 
-  implicit def ipord: Ordering[IntPair] =
-    Ordering.by(ip => (ip.x, ip.y))
-}
+circle1.area
 ```
 
-It's common to see typeclass instances as [implicit values](implicit-parameters.html), such as `ipord` above, defined in the companion, when following the typeclass pattern. This is because the companion's members are included in the default implicit search for related values.
+The `class Circle` contains the val `radius` which is specific to each instance whereas the `object Circle` contains the method `calculateArea` which is the same for every instance.
 
-## Notes for Java programmers ##
+A common pattern to simplify calls to the companion object's members from the companion class is to import everything:
+```
+import java.time.LocalDateTime;
+import java.time.DayOfWeek
 
-`static` is not a keyword in Scala. Instead, all members that would be static, including classes, should go in a singleton object instead. They can be referred to with the same syntax, imported piecemeal or as a group, and so on.
+object Event {
+  private def _isOnTheWeekend(date: LocalDateTime) = Array(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY).contains(date.getDayOfWeek)
+}
 
-Frequently, Java programmers define static members, perhaps `private`, as implementation aids for their instance members. These move to the companion, too; a common pattern is to import the companion object's members in the class, like so:
+class Event(name: String, date: LocalDateTime) {
+  import Event._
 
+  def isOnTheWeekend: Boolean = _isOnTheWeekend(date)
+}
+
+val event = new Event("Dotty Days", LocalDateTime.of(2020, 2, 2, 9, 0, 0))
+event.isOnTheWeekend  // true
 ```
-class X {
-  import X._
 
-  def blah = foo
-}
+The `class Event` imports `_isOnTheWeekend` from the `object Event`. To make a member inaccessible from outside, use `private[this]`.
+
+Note: If a class or object has a companion, both must be defined in the same file. To define them in the REPL, you must enter `:paste` and then paste in the class and companion object code.
+
+## Sharing global state
```

Comment:
This is such a bad practice I'm not sure it's a good idea to give it as an example.  If yes, I'd change the tone and make it "Beware of global state", and end with a thread-safe example.